[jeremiahlzz] iP

jeremiahlzz opened the pull request:
# a heading

## a bullet list
- 1
- 2
- 3

## a numbered list
1. 1
2. 2
3. 3

## a fenced code block (with syntax highlighting)
```java
public static void main(String[] args) {
    // Do Nothing
}
```

## a task list
- [ ] task a
- [x] task b
- [ ] task c

## an emoji
🐀 🐂 🐯 🐰 🐉 🐍 🐴 🐐 🐒 🐓 🐶 🐷 

## a blockquote
> [!Note]
> This is a blockquote with an alert

## a hyperlink
[GitHub GFMD guide](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)

## inline code
This is some inline code: `int i = 0;`.

## some text formatting: bold, italic, strikethrough etc.
**Lorem** _Ipsum_ <sup>small words</sup>

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jeremiahlzz committed:
Level 0. Rename, Greet, Exit

jeremiahlzz committed:
Level 1. Echo

jeremiahlzz committed:
Level 2. Add, List

jeremiahlzz committed:
Level-3: Mark as Done

jeremiahlzz committed:
Level-4: ToDo, Event, Deadline

jeremiahlzz committed:
A-TextUiTesting: Automated Text UI Testing

jeremiahlzz committed:
Level-5: Handle Errors

jeremiahlzz committed:
Level-6: Delete

jeremiahlzz committed:
Moved most message text to enums

jeremiahlzz committed:
Refactor ToDo, Deadline, Event case in Zero.java

jeremiahlzz committed:
Add save task data to file

jeremiahlzz committed:
Merge branch 'branch-Level-7'

jeremiahlzz committed:
Add LocalDateTime support for Tasks

jeremiahlzz committed:
Merge branch 'branch-Level-8'

jeremiahlzz committed:
Refactor Ui, Storage, TaskList, Parser, Commands to individual classes

jeremiahlzz committed:
Merge branch 'branch-A-MoreOOP'

jeremiahlzz committed:
Refactor class structure to follow MVC architecture

jeremiahlzz committed:
Merge branch 'branch-A-Packages'

jeremiahlzz committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

jeremiahlzz committed:
Add gradle support

jeremiahlzz committed:
Merge branch 'branch-A-Gradle'

jeremiahlzz committed:
Add JUnit tests for Parser and TaskList

jeremiahlzz committed:
Merge branch 'branch-A-JUnit'

jeremiahlzz committed:
Add JavaDoc for all classes

jeremiahlzz committed:
Add checkstyle to gradle

delishad21 commented on src/main/java/controller/Zero.java:
```diff
@@ -0,0 +1,164 @@
+                } catch (IndexOutOfBoundsException e) {
+                    ui.showIndexOutOfBoundsError(storage.getTaskList());
+                }
+                break;
```
Can consider abstracting the code for each case into new functions as your run code looks a little crowded. Maybe can also consider nesting the entire switch case in a try block and catching all the exceptions after the switch case has run? Since the exceptions from most of your cases are the same.

jeremiahlzz replied:
Abstracting out each switch case as a function is a great idea! 😄 I will definitely consider it in future projects to improve readability.
Catching all exceptions after the switch case could potentially run into problems in the future, if I want a different UI response for each command. That being said its great idea for the current scope of the project 👍

delishad21 commented on src/main/java/model/ToDo.java:
```diff
@@ -0,0 +1,29 @@
+    private final String name;
+    private final boolean done;
+
+    ToDo(String name) {
```
Missing access modifier here for constructor. Is the constructor meant to be package-private?

jeremiahlzz replied:
Thanks for catching it!

delishad21 commented on src/main/java/view/Ui.java:
```diff
@@ -0,0 +1,97 @@
+
+    public Ui(Reader input, Writer output, String d) {
+        br = new BufferedReader(input);
+        pw = new PrintWriter(output) {
+            @Override
+            public PrintWriter printf(String format, Object... args) {
+                return super.printf(divider + "\n" + format + "\n" + divider + "\n", args);
+            }
+        };
```
This is an interesting and useful way to set up bot responses!

delishad21 commented on src/test/java/controller/ParserTest.java:
```diff
@@ -0,0 +1,32 @@
+    @Test
+    public void parseInput_deadlineTaskInput_success() {
+        Hashtable<String, String> dict = Parser.parseInput("deadline testinput /by 21/2/23 1300");
+        assertAll("test",
+            () -> assertEquals("deadline", dict.get("cmd")),
+            () -> assertEquals("testinput", dict.get("name")),
+            () -> assertEquals("21/2/23 1300", dict.get("/by"))
+        );
```
Very clean testing code. Easy to understand

delishad21 reviewed (commented):
Very neat code that is well thought out and organised into appropriate packages

jeremiahlzz committed:
Add find command

jeremiahlzz committed:
Merge branch 'branch-A-JavaDoc'

jeremiahlzz committed:
Merge branch 'branch-A-CodingStandard'

jeremiahlzz committed:
Merge branch 'branch-Level-9'

jeremiahlzz committed:
Fix some coding violations

jeremiahlzz committed:
Merge branch 'branch-A-CheckStyle'

jeremiahlzz committed:
Add JavaFx GUI

jeremiahlzz committed:
Merge branch 'branch-Level-10'

jeremiahlzz committed:
Add exit animation

jeremiahlzz committed:
Add varargs to TaskList.java

jeremiahlzz committed:
Merge branch 'branch-A-Varargs'

jeremiahlzz committed:
Add assertions to Zero.java

jeremiahlzz committed:
Merge pull request #2 from jeremiahlzz/branch-A-Assertions

Add assertions to Zero.java

jeremiahlzz committed:
Merge branch 'master' into branch-A-CodeQuality

jeremiahlzz committed:
Refactor according code quality feedback

jeremiahlzz committed:
Merge pull request #3 from jeremiahlzz/branch-A-CodeQuality

Branch a code quality

jeremiahlzz committed:
Merge branch 'master' into branch-A-Streams

jeremiahlzz committed:
Add java streams to TaskList.java

jeremiahlzz committed:
Add java streams to TaskList.java

All methods are implemented imperatively using for loops and if-else
statements.

Utilize Java streams to make the match function declarative.

Declarative programming is easier to grasp intuitively and should
result in less logical errors.

jeremiahlzz committed:
Merge pull request #4 from jeremiahlzz/branch-A-Streams

Branch a streams

jeremiahlzz committed:
Fix minor coding violations

jeremiahlzz committed:
Fix minor coding violations

jeremiahlzz committed:
Add tag command

jeremiahlzz committed:
Add tag command

Users are unable to tag their tasks. This can be troubling when
keeping track of similar tasks.

Provide a way to tag items e.g., tag a task as #fun.
Tags can be filtered and displayed in future functionality.

Tags are implemented as a string and only 1 tag is allowed per task.
Tags can be identified with a #hash beside the task name.

jeremiahlzz committed:
Merge branch 'master' into branch-C-Tagging

jeremiahlzz committed:
Add title

jeremiahlzz committed:
Add user guide

jeremiahlzz committed:
Add table of contents
